app/assets/scripts/map.js

```diff
@@ -4,9 +4,68 @@
 // <script src="https://maps.googleapis.com/maps/api/js?key=YOUR_API_KEY&libraries=places">
 
 var map;
+var oms;
 var markers = [];
 var infowindow;
 
+// Click listener for a marker.
+function markerListener(marker, event) {
```

Comment:
No, I don't think we should use a variable instead of a string literal for 'click'. I did modify 251.